src/main/java/gov/cabinetofice/gapuserservice/service/encryption/AESService.java

```diff
@@ -0,0 +1,125 @@
+package gov.cabinetofice.gapuserservice.service.encryption;
+
+import gov.cabinetofice.gapuserservice.dto.EncryptedResponseDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.crypto.*;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import javax.xml.bind.DatatypeConverter;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Optional;
+
+@Service
+public class AESService
+{
+    private static final Logger logger = LoggerFactory.getLogger(AESService.class);
+    private static final int AES_KEY_SIZE = 256;
+    private static final int GCM_IV_LENGTH = 12;
+    private static final int GCM_TAG_LENGTH = 128;
+    private static final String KEY_SELECTOR = "AES";
+    private static final String CIPHER_SELECTOR = "AES/GCM/NoPadding";
+
+    private SecretKey secretKey;
+
+    @Autowired
+    private SigningService signingService;
+
+    @Autowired
+    private Sha512Service sha512Service;
+
+    public AESService() throws Exception {
+        KeyGenerator keyGenerator = KeyGenerator.getInstance(KEY_SELECTOR);
+        keyGenerator.init(AES_KEY_SIZE);
+
+        secretKey = keyGenerator.generateKey();
+
+    }
+
+    public Optional<EncryptedResponseDto> encrypt(String message) {
+        try {
+            byte[] initVector = new byte[GCM_IV_LENGTH];
+            GenerateInitVector(initVector);
+
+            Cipher cipher = Cipher.getInstance(CIPHER_SELECTOR);
+            SecretKeySpec keySpec = new SecretKeySpec(secretKey.getEncoded(), KEY_SELECTOR);
+            GCMParameterSpec gcmSpec = new GCMParameterSpec(GCM_TAG_LENGTH , initVector);
+            cipher.init(Cipher.ENCRYPT_MODE, keySpec, gcmSpec);
+            byte[] cipherText = cipher.doFinal(message.getBytes());
+
+            EncryptedResponseDto response = new EncryptedResponseDto();
+            response.setEncryptedText(DatatypeConverter.printHexBinary(cipherText));
+            response.setIV(initVector);
+            response.setPrivateKey(secretKey);
+            response = signingService.signMessage(response);
+            response.setShaSum(sha512Service.getSHA512SecurePassword(response.getEncryptedText() + response.getSignature()));
+            return Optional.of(response);
+        }
+        catch (NoSuchAlgorithmException
+                | InvalidKeyException
+                | InvalidAlgorithmParameterException
+                | NoSuchPaddingException
+                | BadPaddingException
+                | IllegalBlockSizeException e) {
+            e.printStackTrace();
```

Comment:
Should catch this exception and log it using the Logger mechanism rather than just dumping the stack trace

Reply:
I've removed AESService entirely since we're only hashing now